Sources/MediaNavigationBar.swift

```diff
@@ -0,0 +1,134 @@
+        label.text = NSLocalizedString("TITLE", comment: "Video Title")
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+    
+    lazy var chromeCastButton: UIButton = {
+        var chromeButton = UIButton(type: .system)
+        chromeButton.addTarget(self, action: #selector(toggleChromeCast), for: .touchUpInside)
+        chromeButton.setImage(UIImage(named: "renderer"), for: .normal)
+        chromeButton.tintColor = .white
+        chromeButton.translatesAutoresizingMaskIntoConstraints = false
+        return chromeButton
+    }()
+    
+    @available(iOS 11.0, *)
+    lazy var airPlayRoutePickerViewButton: AVRoutePickerView = {
+        var airPlayRoutePicker = AVRoutePickerView(frame: .zero)
+        airPlayRoutePicker.activeTintColor = .orange
+        airPlayRoutePicker.tintColor = .white
+        airPlayRoutePicker.translatesAutoresizingMaskIntoConstraints = false
+        return airPlayRoutePicker
+    }()
+    
+    private func setupConstraints() {
+        let margin: CGFloat = 35.0
+        
+        var airplayConstraints: [NSLayoutConstraint]
+        if #available(iOS 11.0, *) {
+            airplayConstraints = [
+                chromeCastButton.trailingAnchor.constraint(equalTo: airPlayRoutePickerViewButton.leadingAnchor, constant: margin),
+                airPlayRoutePickerViewButton.trailingAnchor.constraint(equalTo: trailingAnchor, constant: margin),
+                airPlayRoutePickerViewButton.topAnchor.constraint(equalTo: topAnchor)
+            ]
+        } else {
+            airplayConstraints = [
+                chromeCastButton.trailingAnchor.constraint(equalTo: trailingAnchor, constant: margin)
+            ]
+        }
+        
+        let constraints: [NSLayoutConstraint] = [
+            minimizePlaybackButton.leadingAnchor.constraint(equalTo: leadingAnchor),
+            minimizePlaybackButton.topAnchor.constraint(equalTo: topAnchor),
+            minimizePlaybackButton.bottomAnchor.constraint(equalTo: bottomAnchor),
+            mediaTitleTextLabel.leadingAnchor.constraint(equalTo: minimizePlaybackButton.trailingAnchor, constant: margin),
+            mediaTitleTextLabel.topAnchor.constraint(equalTo: topAnchor),
+            mediaTitleTextLabel.trailingAnchor.constraint(lessThanOrEqualTo: chromeCastButton.leadingAnchor),
+            chromeCastButton.topAnchor.constraint(equalTo: topAnchor)
+        ]
+        
+        NSLayoutConstraint.activate(constraints + airplayConstraints)
+    }
+    
+    private func setupViews() {
+        addSubview(minimizePlaybackButton)
+        addSubview(mediaTitleTextLabel)
+        addSubview(chromeCastButton)
+        if #available(iOS 11.0, *) {
+             addSubview(airPlayRoutePickerViewButton)
+        }
+    }
+    
+    // MARK: Initializers
+    required init(coder: NSCoder) {
+        fatalError("init(coder: NSCoder) not implemented")
+    }
+    
+    init() {
```

Comment:
overwrite the designated initializer here instead and call it with  .zero from movieviewcontroller later